File: src/com/fenghuo/notes/adapter/KindAdapter.java
package com.fenghuo.notes.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.Button;
import android.widget.TextView;

import com.fenghuo.notes.db.DbKindHelper;
import com.fenghuo.notes.bean.Kind;
import com.fenghuo.notes.R;

import java.util.List;

public class KindAdapter extends BaseAdapter implements OnClickListener{

	private List<Kind> data;
	private LayoutInflater inflater;
	private Context context;
	private DbKindHelper kindHelper;

	public KindAdapter(Context context, List<Kind> list) {
		inflater = LayoutInflater.from(context);
		this.context = context;
		this.data = list;
	}

	@Override
	public int getCount() {

		return data.size();
	}

	@Override
	public Object getItem(int position) {

		return data.get(position);
	}

	@Override
	public long getItemId(int position) {

		return position;
	}

	@Override
	public View getView(int position, View view, ViewGroup arg2) {

		ViewHolder h = null;
		if (view != null) {
			h = (ViewHolder) view.getTag();
		} else {
			h = new ViewHolder();
			view = inflater.inflate(R.layout.item_kind, null);
			h.item_text = (TextView) view.findViewById(R.id.kind_item_tv);
			h.item_btn_delete = (Button) view.findViewById(R.id.kind_item_delete);
			h.item_btn_delete.setTag(position);
			h.item_btn_delete.setOnClickListener(this);
			view.setTag(h);
		}
		h.item_text.setText(data.get(position).getKind());
		return view;
	}

	private class ViewHolder {
		public TextView item_text;
		public Button item_btn_delete;
	}

	@Override
	public void onClick(View view) {
		int position = Integer.valueOf(view.getTag().toString());
		// 从列表中移除
		kindHelper=new DbKindHelper(context);
		kindHelper.Delete(data.get(position).getId());
		// 从集合中移除
		data.remove(position);
		notifyDataSetChanged();
	}
}

File: src/com/fenghuo/notes/adapter/KindAdapter.kt
package com.fenghuo.notes.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.View.OnClickListener
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.Button
import android.widget.TextView

import com.fenghuo.notes.db.DbKindHelper
import com.fenghuo.notes.bean.Kind
import com.fenghuo.notes.R

class KindAdapter(private val context: Context, private val data: MutableList<Kind>) : BaseAdapter(), OnClickListener {
    private val inflater: LayoutInflater
    private var kindHelper: DbKindHelper? = null

    init {
        inflater = LayoutInflater.from(context)
    }

    override fun getCount(): Int {

        return data.size
    }

    override fun getItem(position: Int): Any {

        return data[position]
    }

    override fun getItemId(position: Int): Long {

        return position.toLong()
    }

    override fun getView(position: Int, view: View?, arg2: ViewGroup): View {
        var view = view

        var h: ViewHolder? = null
        if (view != null) {
            h = view.tag as ViewHolder
        } else {
            h = ViewHolder()
            view = inflater.inflate(R.layout.item_kind, null)
            h.item_text = view!!.findViewById<TextView>(R.id.kind_item_tv) as TextView
            h.item_btn_delete = view.findViewById<Button>(R.id.kind_item_delete) as Button
            h.item_btn_delete!!.tag = position
            h.item_btn_delete!!.setOnClickListener(this)
            view.tag = h
        }
        h.item_text!!.text = data[position].kind
        return view
    }

    private inner class ViewHolder {
        var item_text: TextView? = null
        var item_btn_delete: Button? = null
    }

    override fun onClick(view: View) {
        val position = Integer.valueOf(view.tag.toString())!!
        // 从列表中移除
        kindHelper = DbKindHelper(context)
        kindHelper!!.Delete(data[position].id)
        // 从集合中移除
        data.removeAt(position)
        notifyDataSetChanged()
    }
}
